l1-contracts/test/portals/TokenPortal.sol

```diff
@@ -74,13 +78,16 @@ contract TokenPortal {
     DataStructures.L1ToL2Msg memory message = DataStructures.L1ToL2Msg({
       sender: l1Actor,
       recipient: l2Actor,
-      content: Hash.sha256ToField(abi.encodeWithSignature("mint(uint256,bytes32)", _amount, _to)),
+      content: Hash.sha256ToField(
+        abi.encodeWithSignature("mint(uint256,bytes32)", _amount, _to, msg.sender)
+        ),
       secretHash: _secretHash,
       deadline: _deadline,
       fee: _fee
     });
-    // @todo: (issue #740) implement secure way to cancel the message.
     bytes32 entryKey = inbox.cancelL2Message(message, address(this));
+    // release the funds (since the content hash (& message key) is derived by hashing the caller,
+    // we confirm that cancellor is same as person who created the message)
```

Comment:
If passing in a canceller, the comment here should probably be updated, as it is not needed for the canceller to be the creator as well.